### ISSUE TITLE
The RBM/AutoEncoder layer issue on Spark

### ISSUE THREAD

**Issue by yeseunsu** [deeplearning4j/deeplearning4j#1617]
I'm trying to classify Iris data set using different types of NN layer on Spark local/cluster.
I have not tested all the layers yet but it seems RBM and AutoEncoder are not working on Spark.
When using RBM/AutoEncoder as a layer, the output accuracy score is always 0.3333.
The example works fine on non-spark environment. In this case, the accuracy is something like 0.933.
Is it possible using RBM/AutoEncoder as a layer on Spark environment? 

//The following a simple configuration using RBM 
MultiLayerConfiguration conf = new NeuralNetConfiguration.Builder()
                .seed(123)
                .iterations(iterations)
                .learningRate(1)
                .list()
                .backprop(true)
                .pretrain(false)
                .layer(0, new RBM.Builder().nIn(4).nOut(5).build())
                .layer(1, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD).nIn(5).nOut(3).activation("softmax").build())
                .build();

**Comment by agibsonccc** [comment]
Not sure how to answer this I guess? http://www.slideshare.net/agibsonccc/brief-introduction-to-distributed-deep-learning ANYTHING you do tuning wise is empirical.
